nmrglue/fileio/bruker.py

```diff
@@ -1193,6 +1223,11 @@ def read_pdata(dir=".", bin_files=None, procs_files=None, read_procs=True,
         else:
             raise IOError("No Bruker binary file could be found in %s" % (dir))
 
+    for f in bin_files[1:]:
```

Comment:
This change is not described in the commit message (but is definitely a bug that needs fixing), and is also likely incorrect. I am guessing you wanted to check for the case where `all_components=True`, but the binaries containing imaginary part of the dataset have been deleted. However, this causes a bug when `bin_files` are given manually and only some of the files actually exist. My suggestion would be to loop over all items and check if they exist and remove non existing items using `bin_files.remove(file_that_does_not_exist)` (this duplicates some tests from the previous section, but I guess that is OK),.

Reply:
You are right! This was a hack when I checked on a pure  adsorption spectra from the examdata/exam2d-HH from topspin example (pure adsorption) but I forgot to mention it in the PR. I'll do the suggested change.